### PR TITLE
GTEST: failed make test in case if gtest was disabled

### PR DESCRIPTION
**Pull request by hoopoepg**
fixes https://github.com/openucx/ucx/issues/2939

### CODE REVIEW

File: test/gtest/Makefile.am [hunk 1]
```diff
@@ -208,6 +208,7 @@ noinst_HEADERS = \
 	ucp/ucp_test.h \
 	ucp/ucp_datatype.h
 
+if HAVE_GTEST
```
Comment (yosefe) [inline test/gtest/Makefile.am]: maybe in configure.ac take out this whole makefile out of AC_CONFIG_FILES in case gtest is not enabled?

Comment (hoopoepg) [reply test/gtest/Makefile.am]: it is more complicated, but should work

File: configure.ac [hunk 2]
```diff
@@ -285,12 +285,14 @@ AC_CONFIG_FILES([
                  test/apps/Makefile
                  test/apps/sockaddr/Makefile
                  test/examples/Makefile
-                 test/gtest/Makefile
-                 test/gtest/ucm/test_dlopen/Makefile
                  test/mpi/Makefile
                  bindings/java/src/main/native/Makefile
                  ])
-
+AM_COND_IF([HAVE_GTEST],
```
Comment (yosefe) [inline configure.ac]: AS_IF(["X$enable_gtest" == "Xyes" ] , ...

Comment (hoopoepg) [reply configure.ac]: doesn't work too on distclean
original commit is just best, rolled back to it

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7915/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7917/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7916/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7920/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7925/ for details (Mellanox internal link).

**Pull request merged**